### ISSUE TITLE
ScaleMode doesn't affect the results of globalToLocal

### ISSUE THREAD

**Issue by muchitto** [HeapsIO/heaps#808]
It seems the `s2d.Scene.ScaleMode` doesn't affect the results of the `globalToLocal`-method for instance.

This is a simple example of what I'm attempting here:
```haxe
import h2d.Tile;

class Main extends hxd.App {
    var g2 : h2d.Graphics;

    override function init() {
        var width = 320;
        var height = 240;

        // Just a background color
        var g = new h2d.Graphics(s2d);
        g.beginFill(0x999999);
        g.drawRect(0, 0, width, height);
        g.endFill();

        // The white box
        g2 = new h2d.Graphics(s2d);
        g2.beginFill(0xFFFFFF);
        g2.drawRect(0, 0, 50, 50);
        g2.endFill();

        g2.x = g2.y = 50;

        s2d.scaleMode = LetterBox(width, height, true);

        hxd.Window.getInstance().addEventTarget(onEvent);
    }

    function onEvent (evt : hxd.Event) {
        switch (evt.kind) {
            case EMove:
                /*
                * As the mouse position is the absolute position inside the window,
                * I want to map it to relation to that whitebox (g2 object), but it seems
                * it doesn't take into account the different scalemodes and the changes in the
                * views position and scale?
                * 
                * So when you put your mouse cursor in the top left corner of the whitebox,
                * it should read {0, 0}, right?
                */
                trace(g2.globalToLocal(new h2d.col.Point(evt.relX, evt.relY)));
            case _:
        }
    }

    public static function main () {
        new Main();
    }
}
```

I'm using Heaps from git and HLSDL and Hashlink.

**Comment by Yanrishatum** [comment]
It should not affect it in the first place, because it's whole point in not affecting internal coordinate space and allow flexible adaptability to any screen resolution. Your sample is faulty, because you misunderstand the coordinates involved. You've added event handler to Window, resulting in event giving you **Window coordinates**, not **Scene coordinates**. To obtain scene coordinates, you either have to call private method `Scene.screenToViewport`, or attach your listener to the scene with `addEventListener`, in which case you'll get coordinates in scene coordinate space. There is an important caveat to later approach: Those listeners are fired **only** when no Interactive already processed the even and did not propagate it further.

In short: 
1. It's working as it supposed to do. 
2. You're using it wrong and expect things you should not.
3. Either do `@:privateAccess s2d.screenToViewport(evt);` before accessing `relX`/`relY` or use `s2d.addEventListener` to obtain proper s2d coordinate space instead of window coordinate space.

**Comment by muchitto** [comment]
Okay, makes more sense, I guess. It didn't occur to me that I should just bypass access control and just fiddle around with private functions in the 2D Scene class, because... well... I guess I thought they were private for a reason and I would just mess up some internal workings, or something. Also I just didn't actually notice that function tbh. :) 

That distinction between the different coordinate systems is good to know, because I didn't actually notice you can add events into scenes as well (I was just using the Window `addEventTarget`, because that is what is mentioned in the documentations event section, but I guess that is not updated that often).

I have some interactives that partially cover the areas, I want coordinates from, so that latter example is not possible for me, but I will try that first approach, thanks!
